src/shogun/features/DotFeatures.cpp

```diff
@@ -345,10 +349,13 @@ void CDotFeatures::benchmark_dense_dot_range(int32_t repeats)
 	float64_t* w= SG_MALLOC(float64_t, d);
 	float64_t* out= SG_MALLOC(float64_t, num);
 	float64_t* alphas= SG_MALLOC(float64_t, num);
+#ifdef HAVE_LINALG_LIB
```

Comment:
I suggest you keep the guards for now. But please get rid of this NATIVE thing, we will drop that anyways and eigen3 should be the default.